### PR TITLE
Kraken: In car we go to a parking and we park in it

### PR DESCRIPTION
**Pull request by kinnou02**

### CODE REVIEW

**Comment by kinnou02** [comment]
jenkins, test this please

File: source/type/pb_converter.cpp [hunk 1]
```diff
@@ -815,6 +815,22 @@ const navitia::georef::POI* get_nearest_bss_station(const navitia::type::Data& d
     return vls;
 }
 
+const navitia::georef::POI* get_nearest_parking(const navitia::type::Data& data, const nt::GeographicalCoord& coord) {
+    const navitia::georef::POI* parking = nullptr;
+    //we loop through all poi near the coord to find a vls station within 50 meter
```
Comment (TeXitoi) [inline source/type/pb_converter.cpp]: This look like a bad copy and paste. Please write a get_nearest_poi requiring the poi type and use it.

File: source/type/pb_converter.cpp [hunk 2]
```diff
@@ -815,6 +815,22 @@ const navitia::georef::POI* get_nearest_bss_station(const navitia::type::Data& d
...
+    for (const auto pair: data.geo_ref->poi_proximity_list.find_within(coord, 500)) {
+        const auto poi_idx = pair.first;
+        const auto poi = data.geo_ref->pois[poi_idx];
+        const auto poi_type = data.geo_ref->poitypes[poi->poitype_idx];
+        if (poi_type->uri == "poi_type:amenity:parking") {
+            parking = poi;
+            break;
```
Comment (TeXitoi) [inline source/type/pb_converter.cpp]: I personally prefere early return in this case.

File: source/type/pb_converter.cpp [hunk 3]
```diff
@@ -902,6 +924,14 @@ pbnavitia::Section* create_section(EnhancedResponse& response, pbnavitia::Journe
             LOG4CPLUS_TRACE(log4cplus::Logger::getInstance("logger"), "impossible to find the associated BSS rent station poi for coord " << first_item.coordinates.front());
         }
     }
+    if (first_item.transportation == georef::PathItem::TransportCaracteristic::CarLeaveParking) {
```
Comment (TeXitoi) [inline source/type/pb_converter.cpp]: a switch seems better here

File: source/type/pb_converter.cpp [hunk 4]
```diff
@@ -838,18 +854,24 @@ void finalize_section(pbnavitia::Section* section, const navitia::georef::PathIt
...
         const auto vls_station = get_nearest_bss_station(data, item.coordinates.front());
         if (vls_station) {
             fill_pb_placemark(vls_station, data, dest_place, depth, now, action_period);
-            poi_found = true;
         } else {
-            LOG4CPLUS_TRACE(log4cplus::Logger::getInstance("logger"), "impossible to find the associated BSS putback station poi for coord " << last_item.coordinates.front());
+            LOG4CPLUS_DEBUG(log4cplus::Logger::getInstance("logger"), "impossible to find the associated BSS putback station poi for coord " << last_item.coordinates.front());
+        }
+    }
+    if (item.transportation == georef::PathItem::TransportCaracteristic::CarPark) {
```
Comment (TeXitoi) [inline source/type/pb_converter.cpp]: switch seems better here.

File: source/type/pb_converter.cpp [hunk 5]
```diff
@@ -838,18 +854,24 @@ void finalize_section(pbnavitia::Section* section, const navitia::georef::PathIt
     //add the destination as a placemark
     pbnavitia::PtObject* dest_place = section->mutable_destination();
 
-    bool poi_found = false;
```
Comment (TeXitoi) [inline source/type/pb_converter.cpp:841]: Great!

**Comment by TeXitoi** [comment]
We have poi for CarLeaveParking and BssPutBack, but why not for CarPark and BssTake?

File: source/type/pb_converter.cpp [hunk 6]
```diff
@@ -838,22 +850,35 @@ void finalize_section(pbnavitia::Section* section, const navitia::georef::PathIt
...
     // we want to have a specific place mark for vls or for the departure if we started from a poi
-    if (item.transportation == georef::PathItem::TransportCaracteristic::BssPutBack) {
-        const auto vls_station = get_nearest_bss_station(data, item.coordinates.front());
-        if (vls_station) {
-            fill_pb_placemark(vls_station, data, dest_place, depth, now, action_period);
-            poi_found = true;
-        } else {
-            LOG4CPLUS_TRACE(log4cplus::Logger::getInstance("logger"), "impossible to find the associated BSS putback station poi for coord " << last_item.coordinates.front());
+    switch(item.transportation){
+        case georef::PathItem::TransportCaracteristic::BssPutBack:
```
Comment (antoine-de) [inline source/type/pb_converter.cpp:855]: hum you take only bssPutBack into account here and CarPark and CarLeaveParking after. why ?

**Comment by antoine-de** [comment]
OK for me

**Comment by antoine-de** [comment]
jenkins, test this please

**Comment by kinnou02** [comment]
jenkins, test this please

**Pull request merged**